src/api/controllers/db-controller.ts

```diff
@@ -938,6 +940,8 @@ function parseContractsWithDbTxs(contracts: DbSmartContract[], dbTxs: DbTx[]): T
       if (transaction) {
         transactions.push(transaction as Transaction);
       }
+    } else if (dbTx.type_id !== DbTxTypeId.ContractCall) {
```

Comment:
I'm confused on the condition here. What happens if there is no `contract` and the typeID not a contract call? There's no `else` to handle that, so the tx is just dropped?

Reply:
I was confused about whether it would be mandatory for every transaction of type `ContractCall` to have the associated contract in the DB or not. So, if such a situation occurs, I drop the transaction.
Do you suggest parsing that transaction like all others and passing it on, without the contract information?